### ISSUE TITLE
Keel gets out of sync with reality when instances fail to launch during deployment

### ISSUE THREAD

**Issue by erikmunson** [spinnaker/keel#1401]
tl;dr In scenarios where a deployment happens but instances fail to come up, Keel marks versions as deployed when they are not from a traffic/health perspective. Real life scenario from a recent event:

1. Keel tells Orca to deploy a new server group, but instances aren't coming up due to a capacity shortage
2. Orca sits and waits for instances to come up until it times out, failing the task
3. There are now two active server groups: the old one with healthy instances serving traffic, and the new one which has zero instances of any kind. From a user's perspective this deployment has failed and wasn't cleaned up
4. Keel comes back around, sees there's an "active" server group in the cluster with the right version and the right capacity settings, and marks the version it was trying to deploy as deployed successfully. It does not take into account that there is an old server group serving traffic, or the fact that no instances are up in the new server group

Things were left in a state where the new version was not deployed and the task itself failed, but Keel said everything was great and the new version replaced the old version. Seems very related to the [two active server groups GH issue](https://github.com/spinnaker/keel/issues/1084) we're also aware of.

In the best case, this is unexpected and folks are confused/annoyed that they're seeing incorrect data. In the worst case this could theoretically result in outages/incidents due to either old code running without people realizing _or_ new code coming online as instances are launched later (in this example when the capacity shortage is over), effectively causing multiple versions to run simultaneously without anyone being alerted that things are changing/deploying.

**Comment by emjburns** [comment]
This is done. We now only publish a "deployed" event when the server group is actually healthy - that means that it has all the correct capacity and that instances are "up" if you want them to be up.
